src/transaction/sign.js

```diff
@@ -6,23 +6,38 @@ const binary = require('ripple-binary-codec');
 const {computeBinaryTransactionHash} = require('ripple-hashes');
 const validate = utils.common.validate;
 
-function computeSignature(txJSON, privateKey) {
-  const signingData = binary.encodeForSigning(txJSON);
+function computeSignature(tx: Object, privateKey: string, signFor: ?string) {
+  const signingData = signFor ?
+    binary.encodeForMultisigning(tx, signFor) : binary.encodeForSigning(tx);
   return keypairs.sign(signingData, privateKey);
 }
 
-function sign(txJSON: string, secret: string
+function sign(txJSON: string, secret: string, options: Object = {}
 ): {signedTransaction: string; id: string} {
   validate.sign({txJSON, secret});
   // we can't validate that the secret matches the account because
   // the secret could correspond to the regular key
 
   const tx = JSON.parse(txJSON);
+  if (tx.TxnSignature || tx.Signers) {
+    throw new utils.common.errors.ValidationError(
+      'txJSON must not contain "TxnSignature" or "Signers" properties');
+  }
+
   const keypair = keypairs.deriveKeypair(secret);
-  if (tx.SigningPubKey === undefined) {
-    tx.SigningPubKey = keypair.publicKey;
+  tx.SigningPubKey = options.signFor ? '' : keypair.publicKey;
```

Comment:
My understanding of the reason for "signFor" is that, for example, multiple accounts could have the same regular key and if something is signed by that regular key, we need to know which account it counts for. How about calling it `signAs`?